src/Form/SmsGatewayForm.php

```diff
@@ -148,11 +160,16 @@ public function buildForm(array $form, FormStateInterface $form_state) {
     ];
 
     if (!$sms_gateway->isNew()) {
-      $form['delivery_report_path'] = [
-        '#type' => 'item',
-        '#title' => $this->t('Delivery report URL'),
-        '#markup' => Url::fromRoute('sms.process_delivery_report', ['sms_gateway' => $sms_gateway->id()], ['absolute' => TRUE])->toString(),
-      ];
+      $url = Url::fromRoute('sms.process_delivery_report', ['sms_gateway' => $sms_gateway->id()])->setAbsolute();
+      $access = $this->accessManager->checkNamedRoute($url->getRouteName(), $url->getRouteParameters());
```

Comment:
Then we should check access against the anonymous user, not the logged in (admin) user.